Athena Driver fails to sync fields for partitioned tables

### Describe the bug

This seems related to metabase/metabase#58976 which references \[metabase/metabase#58441\] (I cant access the linear instance, to validate this).

Metabase Athena schema sync fails for a partitioned Athena table when the partition columns are also returned as normal table columns. During sync, Metabase attempts to create duplicate metabase_field records for the partition columns. The duplicate partition-derived fields have database_type = "identity" and both base_type and effective_type set to nil, which causes Metabase’s application database to reject the insert because BASE_TYPE is non-nullable.

The Athena table itself is queryable and has a valid schema. The failure seems to occur during Metabase metadata sync, not during Athena query execution.

## Athena table schema

Running:

sql DESCRIBE `myschema.mytable`;

Returns:

weather_id       string <br>subcounty_id     string <br>weight           float <br>unused_index     int <br>temporal_key     string

The table is partitioned by:

weather_id <br>temporal_key

note. myschema.mytable is a pyiceberg table accessed with a AWS Glue Metastore catalog.

Metabase attempts to create seven fields even though Athena DESCRIBE shows five fields.

The fields Metabase attempts to insert are:

text 0 weather_id      database_type string    base_type type/Text 1 subcounty_id    database_type string    base_type type/Text 2 weight          database_type float     base_type type/Float 3 unused_index    database_type int       base_type type/Integer 4 temporal_key    database_type string    base_type type/Text 5 weather_id      database_type identity  base_type nil 6 temporal_key    database_type identity  base_type nil

The two failing rows are duplicate entries for the partition columns:

text name = weather_id database_type = identity effective_type = nil base_type = nil

and:

text name = temporal_key database_type = identity effective_type = nil base_type = nil

Metabase then attempts to insert those rows into METABASE_FIELD, but BASE_TYPE is non-nullable, so the insert fails.

## Relevant log excerpt

text WARN sync.util :: Error checking if Fields ("temporal_key" "temporal_key" "subcounty_id" "weather_id" "unused_index" "weight" "weather_id") need to be created or reactivated  clojure.lang.ExceptionInfo: NULL not allowed for column "BASE_TYPE"; SQL statement:  INSERT INTO "METABASE_FIELD" ... \[23502-214\]

The duplicate field list includes:

text temporal_key temporal_key weather_id weather_id

The parsed Metabase field rows show the root cause:

clojure {:database_type "string",  :name "weather_id",  :effective_type :type/Text,  :base_type :type/Text}  {:database_type "string",  :name "temporal_key",  :effective_type :type/Text,  :base_type :type/Text}  {:database_type "identity",  :name "weather_id",  :effective_type nil,  :base_type nil}  {:database_type "identity",  :name "temporal_key",  :effective_type nil,  :base_type nil}

### To Reproduce

1. Create or use an AWS Athena table with normal data columns and partition columns.
2. Include the partition columns in the Athena table schema. Example columns:
   * weather_id string
   * subcounty_id string
   * weight float
   * unused_index int
   * temporal_key string
3. Partition the table by the same columns:
   * weather_id
   * temporal_key
4. Verify the table works in Athena:

sql DESCRIBE `myschema.mytable`;

Expected Athena output:

weather_id       string <br>subcounty_id     string <br>weight           float <br>unused_index     int <br>temporal_key     string

5. Add the Athena database to Metabase.
6. Run schema sync for the Athena database.
7. Observe that Metabase fails while syncing fields for the partitioned table.

### Expected behavior

Metabase should sync the Athena table successfully.

Partition columns should either:

* be represented once as normal fields, with the correct Athena type, or
* be represented as partitioned fields with a valid base_type and effective_type.

Metabase should not create duplicate field records with base_type = nil.

### Logs

2026-06-09 16:21:05.152 | 2026-06-09 23:21:05,151 WARN sync.util :: Error checking if Fields ("temporal_key" "temporal_key" "subcounty_id" "weather_id" "unused_index" "weight" "weather_id") need to be created or reactivated <br>2026-06-09 16:21:05.152 | clojure.lang.ExceptionInfo: NULL not allowed for column "BASE_TYPE"; SQL statement:<br>2026-06-09 16:21:05.152 | INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "DATABASE_IS_PK", "UPDATED_AT", "DATABASE_DEFAULT", "EFFECTIVE_TYPE", "NFC_PATH", "PARENT_ID", "DATABASE_PARTITIONED", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "DATABASE_IS_GENERATED", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "DATABASE_IS_NULLABLE", "CREATED_AT", "BASE_TYPE") VALUES (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL) \[23502-214\] {:toucan2/context-trace \[\["execute SQL with class com.mchange.v2.c3p0.impl.NewProxyConnectionJdbc43Full" {:toucan2.jdbc.query/sql-args \["INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "DATABASE_IS_PK", "UPDATED_AT", "DATABASE_DEFAULT", "EFFECTIVE_TYPE", "NFC_PATH", "PARENT_ID", "DATABASE_PARTITIONED", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "DATABASE_IS_GENERATED", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "DATABASE_IS_NULLABLE", "CREATED_AT", "BASE_TYPE") VALUES (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL)" "string" 31 "weather_id" "auto-list" "type/Text" 0 "normal" "Weather ID" 0 "type/Text" "string" 31 "subcounty_id" "auto-list" "type/Text" 1 "normal" "Subcounty ID" 1 "type/Text" "float" 31 "weight" "type/Float" 2 "normal" "Weight" 2 "type/Float" "int" 31 "unused_index" "auto-list" "type/Integer" 3 "normal" "Unused Index" 3 "type/Integer" "string" 31 "temporal_key" "auto-list" "type/Text" 4 "normal" "Temporal Key" 4 "type/Text" "identity" 31 "weather_id" "auto-list" 5 "normal" "Weather ID" 5 "identity" 31 "temporal_key" "auto-list" 6 "normal" "Temporal Key" 6\]}\] \["resolve connection" {:toucan2.connection/connectable org.h2.jdbc.JdbcConnection}\] \["resolve connection" {:toucan2.connection/connectable nil}\] {:toucan2.pipeline/rf #object\[clojure.core$map$fn__5952$fn__5953 0x4d12bab6 "clojure.core$map$fn__5952$fn__5953@4d12bab6"\]} \["with compiled query" {:toucan2.pipeline/compiled-query \["INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "DATABASE_IS_PK", "UPDATED_AT", "DATABASE_DEFAULT", "EFFECTIVE_TYPE", "NFC_PATH", "PARENT_ID", "DATABASE_PARTITIONED", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "DATABASE_IS_GENERATED", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "DATABASE_IS_NULLABLE", "CREATED_AT", "BASE_TYPE") VALUES (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL)" "string" 31 "weather_id" "auto-list" "type/Text" 0 "normal" "Weather ID" 0 "type/Text" "string" 31 "subcounty_id" "auto-list" "type/Text" 1 "normal" "Subcounty ID" 1 "type/Text" "float" 31 "weight" "type/Float" 2 "normal" "Weight" 2 "type/Float" "int" 31 "unused_index" "auto-list" "type/Integer" 3 "normal" "Unused Index" 3 "type/Integer" "string" 31 "temporal_key" "auto-list" "type/Text" 4 "normal" "Temporal Key" 4 "type/Text" "identity" 31 "weather_id" "auto-list" 5 "normal" "Weather ID" 5 "identity" 31 "temporal_key" "auto-list" 6 "normal" "Temporal Key" 6\]}\] \["with built query" {:toucan2.pipeline/built-query {:insert-into \[:metabase_field\], :values ((toucan2.instance/instance :model/Field {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type "type/Text", :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 0, :visibility_type "normal", :database_is_generated nil, :display_name "Weather ID", :database_position 0, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type "type/Text"}) (toucan2.instance/instance :model/Field {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "subcounty_id", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type "type/Text", :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 1, :visibility_type "normal", :database_is_generated nil, :display_name "Subcounty ID", :database_position 1, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type "type/Text"}) (toucan2.instance/instance :model/Field {:description nil, :database_type "float", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weight", :has_field_values nil, :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type "type/Float", :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 2, :visibility_type "normal", :database_is_generated nil, :display_name "Weight", :database_position 2, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type "type/Float"}) (toucan2.instance/instance :model/Field {:description nil, :database_type "int", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "unused_index", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type "type/Integer", :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 3, :visibility_type "normal", :database_is_generated nil, :display_name "Unused Index", :database_position 3, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type "type/Integer"}) (toucan2.instance/instance :model/Field {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type "type/Text", :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 4, :visibility_type "normal", :database_is_generated nil, :display_name "Temporal Key", :database_position 4, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type "type/Text"}) (toucan2.instance/instance :model/Field {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 5, :visibility_type "normal", :database_is_generated nil, :display_name "Weather ID", :database_position 5, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type nil}) (toucan2.instance/instance :model/Field {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values "auto-list", :database_is_pk nil, :updated_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 6, :visibility_type "normal", :database_is_generated nil, :display_name "Temporal Key", :database_position 6, :database_required false, :database_is_nullable nil, :created_at \[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamp"}\], :base_type nil}))}}\] \["resolve connection" {:toucan2.connection/connectable metabase.app_db.connection.ApplicationDB}\] \["resolve connection" {:toucan2.connection/connectable :default}\] \["resolve connection" {:toucan2.connection/connectable nil}\] \["with resolved query" {:toucan2.pipeline/resolved-query {}}\] \["with parsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.pks, :toucan2.pipeline/parsed-args {:rows \[{:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 0, :visibility_type :normal, :database_is_generated nil, :display_name "Weather ID", :database_position 0, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "subcounty_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 1, :visibility_type :normal, :database_is_generated nil, :display_name "Subcounty ID", :database_position 1, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "float", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weight", :has_field_values nil, :database_is_pk nil, :database_default nil, :effective_type :type/Float, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 2, :visibility_type :normal, :database_is_generated nil, :display_name "Weight", :database_position 2, :database_required false, :database_is_nullable nil, :base_type :type/Float} {:description nil, :database_type "int", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "unused_index", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Integer, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 3, :visibility_type :normal, :database_is_generated nil, :display_name "Unused Index", :database_position 3, :database_required false, :database_is_nullable nil, :base_type :type/Integer} {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 4, :visibility_type :normal, :database_is_generated nil, :display_name "Temporal Key", :database_position 4, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 5, :visibility_type :normal, :database_is_generated nil, :display_name "Weather ID", :database_position 5, :database_required false, :database_is_nullable nil, :base_type nil} {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 6, :visibility_type :normal, :database_is_generated nil, :display_name "Temporal Key", :database_position 6, :database_required false, :database_is_nullable nil, :base_type nil}\]}}\] \["with model" {:toucan2.pipeline/model :model/Field}\] \["with unparsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.pks, :toucan2.pipeline/unparsed-args (:model/Field ({:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 0, :visibility_type :normal, :database_is_generated nil, :display_name "Weather ID", :database_position 0, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "subcounty_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 1, :visibility_type :normal, :database_is_gener2026-06-09T23:21:05.152614923Z ated nil, :display_name "Subcounty ID", :database_position 1, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "float", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weight", :has_field_values nil, :database_is_pk nil, :database_default nil, :effective_type :type/Float, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 2, :visibility_type :normal, :database_is_generated nil, :display_name "Weight", :database_position 2, :database_required false, :database_is_nullable nil, :base_type :type/Float} {:description nil, :database_type "int", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "unused_index", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Integer, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 3, :visibility_type :normal, :database_is_generated nil, :display_name "Unused Index", :database_position 3, :database_required false, :database_is_nullable nil, :base_type :type/Integer} {:description nil, :database_type "string", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type :type/Text, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 4, :visibility_type :normal, :database_is_generated nil, :display_name "Temporal Key", :database_position 4, :database_required false, :database_is_nullable nil, :base_type :type/Text} {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "weather_id", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 5, :visibility_type :normal, :database_is_generated nil, :display_name "Weather ID", :database_position 5, :database_required false, :database_is_nullable nil, :base_type nil} {:description nil, :database_type "identity", :semantic_type nil, :table_id 31, :coercion_strategy nil, :name "temporal_key", :has_field_values :auto-list, :database_is_pk nil, :database_default nil, :effective_type nil, :nfc_path nil, :parent_id nil, :database_partitioned nil, :database_is_auto_increment false, :json_unfolding false, :position 6, :visibility_type :normal, :database_is_generated nil, :display_name "Temporal Key", :database_position 6, :database_required false, :database_is_nullable nil, :base_type nil}))}\]\]}<br>2026-06-09 16:21:05.152 | 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)<br>2026-06-09 16:21:05.152 | 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)<br>2026-06-09 16:21:05.152 | 	at org.h2.message.DbException.get(DbException.java:223)<br>2026-06-09 16:21:05.152 | 	at org.h2.message.DbException.get(DbException.java:199)<br>2026-06-09 16:21:05.152 | 	at org.h2.table.Column.validateConvertUpdateSequence(Column.java:365)<br>2026-06-09 16:21:05.152 | 	at org.h2.table.Table.convertInsertRow(Table.java:926)<br>2026-06-09 16:21:05.152 | 	at org.h2.command.dml.Insert.insertRows(Insert.java:167)<br>2026-06-09 16:21:05.152 | 	at org.h2.command.dml.Insert.update(Insert.java:135)<br>2026-06-09 16:21:05.152 | 	at org.h2.command.CommandContainer.executeUpdateWithGeneratedKeys(CommandContainer.java:242)<br>2026-06-09 16:21:05.152 | 	at org.h2.command.CommandContainer.update(CommandContainer.java:163)<br>2026-06-09 16:21:05.152 | 	at org.h2.command.Command.executeUpdate(Command.java:252)<br>2026-06-09 16:21:05.152 | 	at org.h2.jdbc.JdbcPreparedStatement.execute(JdbcPreparedStatement.java:254)<br>2026-06-09 16:21:05.152 | 	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.execute(NewProxyPreparedStatement.java:223)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.query$reduce_jdbc_query.invokeStatic(query.clj:40)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.query$reduce_jdbc_query.invoke(query.clj:22)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invokeStatic(pipeline.clj:19)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invoke(pipeline.clj:9)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:15)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invokeStatic(pipeline.clj:30)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invoke(pipeline.clj:22)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:15)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__35223$combined_method_thread_last__35224.invoke(threaded.clj:64)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_execute$with_connection_STAR___37392$with_transaction_STAR___37393.invoke(pipeline.clj:75)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:196)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:178)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:13)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:13)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_execute$with_connection_STAR___37392.invoke(pipeline.clj:74)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invokeStatic(connection.clj:13)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invoke(connection.clj:11)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_primary_method\_.invokeStatic(connection.clj:204)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_primary_method\_.invoke(connection.clj:194)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_execute.invokeStatic(pipeline.clj:74)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_execute.invoke(pipeline.clj:64)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.Var.invoke(Var.java:401)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_compiled_query.invokeStatic(pipeline.clj:244)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_compiled_query.invoke(pipeline.clj:240)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_built_query.invokeStatic(pipeline.clj:252)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_built_query.invoke(pipeline.clj:246)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_query_primary_method_default.invokeStatic(pipeline.clj:272)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_query_primary_method_default.invoke(pipeline.clj:269)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:15)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__35223$combined_method_thread_last__35224.invoke(threaded.clj:64)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___41566$with_transaction_STAR___41567.invoke(before_insert.clj:37)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_transaction$thunk__45321.invoke(connection.clj:146)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_transaction.invokeStatic(connection.clj:164)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_transaction.invoke(connection.clj:142)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:205)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:178)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:13)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:13)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)<br>2026-06-09 16:21:05.152 | 	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___41566.invoke(before_insert.clj:36)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$bind_current_connectable_fn$fn__37070.invoke(connection.clj:104)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)<br>2026-06-09 16:21:05.152 | 	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:133)<br>2026-06-09 16:21:05.152 | 	at metabase.app_db.connection$do_with_connection_primary_method_default.invoke(connection.clj:131)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_primary_method\_.invokeStatic(connection.clj:204)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_primary_method\_.invoke(connection.clj:194)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)<br>2026-06-09 16:21:05.152 | 	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:12)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)<br>2026-06-09 16:21:05.152 | 	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invokeStatic(before_insert.clj:36)<br>2026-06-09 16:21:05.152 | 	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invoke(before_insert.clj:31)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.combo.common$partial_STAR\_$fn__34913.invoke(common.clj:15)<br>2026-06-09 16:21:05.152 | 	at methodical.util.FnWithMeta.invoke(util.clj:46)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)<br>2026-06-09 16:21:05.152 | 	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_query_STAR\_.invokeStatic(pipeline.clj:278)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_query_STAR\_.invoke(pipeline.clj:274)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_unparsed_with_default_rf.invokeStatic(pipeline.clj:392)<br>2026-06-09 16:21:05.152 | 	at toucan2.pipeline$transduce_unparsed_with_default_rf.invoke(pipeline.clj:386)<br>2026-06-09 16:21:05.152 | 	at toucan2.insert$insert_returning_pks_BANG\_.invokeStatic(insert.clj:139)<br>2026-06-09 16:21:05.152 | 	at toucan2.insert$insert_returning_pks_BANG\_.doInvoke(insert.clj:126)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.RestFn.invoke(RestFn.java:424)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$insert_new_fields_BANG\_.invokeStatic(sync_instances.clj:47)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$insert_new_fields_BANG\_.invoke(sync_instances.clj:40)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$create_or_reactivate_fields_BANG\_.invokeStatic(sync_instances.clj:106)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$create_or_reactivate_fields_BANG\_.invoke(sync_instances.clj:92)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_active_instances_BANG\_$iter__99131__99135$fn__99136$fn__99141.invoke(sync_instances.clj:141)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:225)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invoke(util.clj:218)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_active_instances_BANG\_$iter__99131__99135$fn__99136.invoke(sync_instances.clj:137)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.force(LazySeq.java:50)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.realize(LazySeq.java:89)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.seq(LazySeq.java:106)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.RT.seq(RT.java:555)<br>2026-06-09 16:21:05.152 | 	at clojure.core$seq__5488.invokeStatic(core.clj:139)<br>2026-06-09 16:21:05.152 | 	at clojure.core$filter$fn__5983.invoke(core.clj:2826)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.force(LazySeq.java:50)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.realize(LazySeq.java:89)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.LazySeq.seq(LazySeq.java:106)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.RT.seq(RT.java:555)<br>2026-06-09 16:21:05.152 | 	at clojure.core$seq__5488.invokeStatic(core.clj:139)<br>2026-06-09 16:21:05.152 | 	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24)<br>2026-06-09 16:21:05.152 | 	at clojure.core.protocols$fn__8266.invokeStatic(protocols.clj:74)<br>2026-06-09 16:21:05.152 | 	at clojure.core.protocols$fn__8266.invoke(protocols.clj:74)<br>2026-06-09 16:21:05.152 | 	at clojure.core.protocols$fn__8207$G__8202__8220.invoke(protocols.clj:13)<br>2026-06-09 16:21:05.152 | 	at clojure.core$reduce.invokeStatic(core.clj:6961)<br>2026-06-09 16:21:05.152 | 	at clojure.core$reduce.invoke(core.clj:6947)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$sum_for_STAR\_.invokeStatic(util.clj:704)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$sum_for_STAR\_.invoke(util.clj:701)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_active_instances_BANG\_.invokeStatic(sync_instances.clj:136)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_active_instances_BANG\_.invoke(sync_instances.clj:123)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_instances_BANG\_.invokeStatic(sync_instances.clj:234)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_instances_BANG\_.invoke(sync_instances.clj:215)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_instances_BANG\_.invokeStatic(sync_instances.clj:221)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields.sync_instances$sync_instances_BANG\_.invoke(sync_instances.clj:215)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_and_update_BANG\_.invokeStatic(fields.clj:66)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_and_update_BANG\_.invoke(fields.clj:60)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_$fn__99603$sync_BANG___99604$fn__99605$fn__99608.invoke(fields.clj:109)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_$fn__99603$sync_BANG___99604$fn__99605.invoke(fields.clj:101)<br>2026-06-09 16:21:05.152 | 	at clojure.core$map$fn__5952$fn__5953.invoke(core.clj:2759)<br>2026-06-09 16:21:05.152 | 	at clojure.core$partition_by$fn__8635$fn__8636.invoke(core.clj:7339)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.PersistentVector.reduce(PersistentVector.java:418)<br>2026-06-09 16:21:05.152 | 	at clojure.core$transduce.invokeStatic(core.clj:7025)<br>2026-06-09 16:21:05.152 | 	at clojure.core$transduce.invoke(core.clj:7012)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_$fn__99603$sync_BANG___99604.invoke(fields.clj:89)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_$fn__99603.invoke(fields.clj:131)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:225)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invoke(util.clj:218)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_.invokeStatic(fields.clj:85)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata.fields$sync_fields_BANG\_.invoke(fields.clj:80)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyToHelper(AFn.java:154)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyTo(AFn.java:144)<br>2026-06-09 16:21:05.152 | 	at clojure.core$apply.invokeStatic(core.clj:669)<br>2026-06-09 16:21:05.152 | 	at clojure.core$apply.invoke(core.clj:662)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_step_with_metadata$fn__72740$fn__72741$fn__72744.invoke(util.clj:616)<br>2026-06-09 16:21:05.152 | 	at metabase.task_history.models.task_history$do_with_task_history.invokeStatic(task_history.clj:238)<br>2026-06-09 16:21:05.152 | 	at metabase.task_history.models.task_history$do_with_task_history.invoke(task_history.clj:221)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_step_with_metadata$fn__72740$fn__72741.invoke(util.clj:609)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_returning_throwable.invokeStatic(util.clj:247)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_returning_throwable.invoke(util.clj:244)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_step_with_metadata$fn__72740.doInvoke(util.clj:608)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.RestFn.invoke(RestFn.java:400)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_start_and_finish_logging_STAR\_.invokeStatic(util.clj:161)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_start_and_finish_logging_STAR\_.invoke(util.clj:155)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_start_and_finish_debug_logging.invokeStatic(util.clj:180)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_start_and_finish_debug_logging.invoke(util.clj:176)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_step_with_metadata.invokeStatic(util.clj:603)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_step_with_metadata.invoke(util.clj:597)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_sync_operation$fn__72791$fn__72799.invoke(util.clj:679)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_sync_operation$fn__72791.invoke(util.clj:677)<br>2026-06-09 16:21:05.152 | 	at metabase.task_history.models.task_history$do_with_task_history.invokeStatic(task_history.clj:238)<br>2026-06-09 16:21:05.152 | 	at metabase.task_history.models.task_history$do_with_task_history.invoke(task_history.clj:221)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_sync_operation.invokeStatic(util.clj:674)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$run_sync_operation.invoke(util.clj:669)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata$sync_db_metadata_BANG__STAR\_.invokeStatic(sync_metadata.clj:69)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata$sync_db_metadata_BANG__STAR\_.invoke(sync_metadata.clj:63)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata$sync_db_metadata_explicit_BANG\_$fn__100613.invoke(sync_metadata.clj:87)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:225)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_with_error_handling.invoke(util.clj:218)<br>2026-06-09 16:21:05.152 | 	at clojure.core$partial$fn__5931.invoke(core.clj:2647)<br>2026-06-09 16:21:05.152 | 	at metabase.driver$fn__58172.invokeStatic(driver.clj:1088)<br>2026-06-09 16:21:05.152 | 	at metabase.driver$fn__58172.invoke(driver.clj:1088)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.MultiFn.invoke(MultiFn.java:239)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$sync_in_context$fn__72593.invoke(util.clj:197)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_db_logging_disabled$fn__72590.invoke(util.clj:189)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_start_and_finish_logging_STAR\_.invokeStatic(util.clj:161)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_start_and_finish_logging_STAR\_.invoke(util.clj:155)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_start_and_finish_logging$fn__72567.invoke(util.clj:174)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_sync_events$fn__72562.invoke(util.clj:147)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$with_duplicate_ops_prevented$fn__72541.invoke(util.clj:119)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_sync_operation_STAR\_.invokeStatic(util.clj:294)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_sync_operation_STAR\_.invoke(util.clj:269)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_explicit_sync_operation.invokeStatic(util.clj:321)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.util$do_explicit_sync_operation.invoke(util.clj:311)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata$sync_db_metadata_explicit_BANG\_.invokeStatic(sync_metadata.clj:86)<br>2026-06-09 16:21:05.152 | 	at metabase.sync.sync_metadata$sync_db_metadata_explicit_BANG\_.invoke(sync_metadata.clj:82)<br>2026-06-09 16:21:05.152 | 	at metabase.warehouses_rest.api$fn__142653$fn__142659$fn__142665$fn__142666.invoke(api.clj:1272)<br>2026-06-09 16:21:05.152 | 	at metabase.database_routing.core$fn__102476$with_database_routing_off_fn__102477.invoke(core.clj:24)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyToHelper(AFn.java:154)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyTo(AFn.java:144)<br>2026-06-09 16:21:05.152 | 	at clojure.core$apply.invokeStatic(core.clj:667)<br>2026-06-09 16:21:05.152 | 	at clojure.core$apply.invoke(core.clj:662)<br>2026-06-09 16:21:05.152 | 	at metabase.premium_features.defenterprise$dynamic_ee_oss_fn$fn__52354.doInvoke(defenterprise.clj:60)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.RestFn.invoke(RestFn.java:411)<br>2026-06-09 16:21:05.152 | 	at metabase.warehouses_rest.api$fn__142653$fn__142659$fn__142665.invoke(api.clj:1270)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)<br>2026-06-09 16:21:05.152 | 	at clojure.lang.AFn.applyTo(AFn.java:144)<br>2026-06-09 16:21:05.152 | 	at clojure.core$apply.invokeStatic(core.clj:667)<br>2026-06-09 16:21:05.152 | 	at clojure.core$with_bindings_STAR\_.invokeStatic(core.clj:1990)<br>2026-06-09 16:21:05.152 | 	at clojure.core$with_bindings_STAR\_.doInvoke(core.clj:1990)<br>2026-06-09 16:21:05.153 | 	at clojure.lang.RestFn.invoke(RestFn.java:428)<br>2026-06-09 16:21:05.153 | 	at clojure.lang.AFn.applyToHelper(AFn.java:156)<br>2026-06-09 16:21:05.153 | 	at clojure.lang.RestFn.applyTo(RestFn.java:135)<br>2026-06-09 16:21:05.153 | 	at clojure.core$apply.invokeStatic(core.clj:671)<br>2026-06-09 16:21:05.153 | 	at clojure.core$bound_fn_STAR\_$fn__5839.doInvoke(core.clj:2020)<br>2026-06-09 16:21:05.153 | 	at clojure.lang.RestFn.invoke(RestFn.java:400)<br>2026-06-09 16:21:05.153 | 	at clojure.lang.AFn.call(AFn.java:18)<br>2026-06-09 16:21:05.153 | 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)<br>2026-06-09 16:21:05.153 | 	at java.base/java.lang.Thread.run(Unknown Source)<br>2026-06-09 16:21:05.153 | Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: NULL not allowed for column "BASE_TYPE"; SQL statement:<br>2026-06-09 16:21:05.153 | INSERT INTO "METABASE_FIELD" ("DESCRIPTION", "DATABASE_TYPE", "SEMANTIC_TYPE", "TABLE_ID", "COERCION_STRATEGY", "NAME", "HAS_FIELD_VALUES", "DATABASE_IS_PK", "UPDATED_AT", "DATABASE_DEFAULT", "EFFECTIVE_TYPE", "NFC_PATH", "PARENT_ID", "DATABASE_PARTITIONED", "DATABASE_IS_AUTO_INCREMENT", "JSON_UNFOLDING", "POSITION", "VISIBILITY_TYPE", "DATABASE_IS_GENERATED", "DISPLAY_NAME", "DATABASE_POSITION", "DATABASE_REQUIRED", "DATABASE_IS_NULLABLE", "CREATED_AT", "BASE_TYPE") VALUES (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, NULL, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, ?, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), ?), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL), (NULL, ?, NULL, ?, NULL, ?, ?, NULL, NOW(), NULL, NULL, NULL, NULL, NULL, FALSE, FALSE, ?, ?, NULL, ?, ?, FALSE, NULL, NOW(), NULL) \[23502-214\]<br>2026-06-09 16:21:05.153 | 	... 267 more

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/147.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "metabase-info": {
    "databases": [
      "athena",
      "h2"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2026-06-08",
      "tag": "v0.62.1.1",
      "hash": "5192f94"
    },
    "settings": {
      "report-timezone": null
    },
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "21.0.11+10-LTS",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "21.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "21.0.11+10-LTS",
    "os.name": "Linux",
    "os.version": "6.10.14-linuxkit",
    "user.language": "en",
    "user.timezone": "GMT"
  }
}
```

### Severity

blocking usage of metabase entirely

### Additional context

Excited about trying out Metabase, unfortunately this is a major blocker. The visual query editor tool would be a great fit for my org.

--- a/modules/drivers/athena/src/metabase/driver/athena.clj
+++ b/modules/drivers/athena/src/metabase/driver/athena.clj
@@ -370,14 +370,32 @@
   {:name (str/trim (:col_name column-metadata))
    :type (str/trim (:data_type column-metadata))})
 
+;; the partition section markers and row shapes come from Hive's DESCRIBE formatter, which this
+;; output format derives from:
+;; https://github.com/apache/hive/blob/1cb455bbb6753e548e1435f4345b4e0d4644bf9f/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/formatter/TextDescTableFormatter.java#L121-L185
+(defn- partition-section-start?
+  "Whether this row of `DESCRIBE` output starts the partition section: `# Partition Information`
+  for natively-partitioned tables, or `# Partition Transform Information` for Iceberg tables."
+  [{:keys [col_name]}]
+  (str/starts-with? (str col_name) "# Partition"))
+
 (defn- remove-invalid-columns
-  "Returns a transducer."
+  "Returns a transducer that converts rows of `DESCRIBE` output into column metadata, dropping the
+  partition section and other non-column rows."
   []
-  (comp (remove #(= (:col_name %) ""))
+  ;; everything from the partition section marker on describes the partition spec rather than
+  ;; additional columns. Hive tables repeat the partition columns there with their real types;
+  ;; Iceberg tables repeat them with the partition transform (e.g. `identity`) as the type, which
+  ;; would otherwise sync as duplicate columns with no base type (#75579)
+  (comp (take-while (complement partition-section-start?))
+        (remove #(= (:col_name %) ""))
         (remove #(= (:col_name %) nil))
         (remove #(= (:data_type %) nil))
         (remove #(str/starts-with? (:col_name %) "#")) ; remove comment
-        (distinct)                                     ; driver can return twice the partitioning fields
+        ;; the partition section only repeats columns that are already in the main column list, so
+        ;; if the marker text above ever changes, keeping the first occurrence of each name still
+        ;; keeps the partition rows out
+        (m/distinct-by :col_name)
         (map describe-database->clj)))
 
 (defn- normalize-field-info

--- a/modules/drivers/athena/src/metabase/driver/athena/schema_parser.clj
+++ b/modules/drivers/athena/src/metabase/driver/athena/schema_parser.clj
@@ -6,8 +6,8 @@
 
 (set! *warn-on-reflection* true)
 
-(defn- column->base-type [column-type]
-  (sql-jdbc.sync/database-type->base-type :athena (keyword (re-find #"\w+" column-type))))
+(defn- column->base-type [column-name column-type]
+  (sql-jdbc.sync/database-type->base-type-or-warn :athena [column-name] (re-find #"\w+" column-type)))
 
 #_(defn- create-nested-fields [schema database-position]
     (set (map (fn [[k v]]
@@ -55,6 +55,6 @@
 
     :else
     {:name              (:name field-info)
-     :base-type         (column->base-type (:type field-info))
+     :base-type         (column->base-type (:name field-info) (:type field-info))
      :database-type     (:type field-info)
      :database-position (:database-position field-info)}))

--- a/modules/drivers/athena/test/metabase/driver/athena_test.clj
+++ b/modules/drivers/athena/test/metabase/driver/athena_test.clj
@@ -59,24 +59,27 @@
              {:name "ts", :base-type :type/Text, :database-type "STRING", :database-position 1}}
            (#'athena/describe-table-fields-without-nested-fields :athena "test" "test" upper-case-schema-columns)))))
 
+;; the partition section format comes from Hive's DESCRIBE formatter; see
+;; `PARTITION_TRANSFORM_SPEC_SCHEMA` in
+;; https://github.com/apache/hive/blob/1cb455bbb6753e548e1435f4345b4e0d4644bf9f/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/DescTableDesc.java#L43
 (def ^:private iceberg-partitioned-schema
-  ;; DESCRIBE output for an Iceberg table partitioned by identity(weather_id) and
-  ;; identity(temporal_key). Unlike Hive tables, where the partition section repeats the columns
-  ;; with their real types, Iceberg lists the partition *transform* (e.g. `identity`) in the type
-  ;; column.
+  "DESCRIBE output for an Iceberg table partitioned by `identity(weather_id)` and
+  `identity(temporal_key)`. Unlike Hive tables, where the partition section repeats the columns
+  with their real types, Iceberg lists the partition transform (e.g. `identity`) in the type
+  column."
   [{:_col0 "weather_id\tstring\t"}
    {:_col0 "subcounty_id\tstring\t"}
    {:_col0 "weight\tfloat\t"}
    {:_col0 "unused_index\tint\t"}
    {:_col0 "temporal_key\tstring\t"}
-   {:_col0 "# Partition Information\t\t"}
+   {:_col0 "# Partition Transform Information\t\t"}
    {:_col0 "# col_name\ttransform_type\t"}
    {:_col0 "weather_id\tidentity\t"}
    {:_col0 "temporal_key\tidentity\t"}])
 
 (def ^:private hive-partitioned-schema
-  ;; The same table as a Hive-style partitioned table: the partition section repeats the partition
-  ;; columns with their real types.
+  "The same table as a Hive-style partitioned table: the partition section repeats the partition
+  columns with their real types."
   [{:_col0 "weather_id\tstring\t"}
    {:_col0 "subcounty_id\tstring\t"}
    {:_col0 "weight\tfloat\t"}
@@ -87,20 +90,38 @@
    {:_col0 "weather_id\tstring\t"}
    {:_col0 "temporal_key\tstring\t"}])
 
-(deftest describe-iceberg-partitioned-table-test
+(deftest ^:parallel describe-iceberg-partitioned-table-test
   (let [expected-fields #{{:name "weather_id", :base-type :type/Text, :database-type "string", :database-position 0}
                           {:name "subcounty_id", :base-type :type/Text, :database-type "string", :database-position 1}
                           {:name "weight", :base-type :type/Float, :database-type "float", :database-position 2}
                           {:name "unused_index", :base-type :type/Integer, :database-type "int", :database-position 3}
                           {:name "temporal_key", :base-type :type/Text, :database-type "string", :database-position 4}}]
-    (testing "Iceberg partition-spec rows in DESCRIBE output must not become phantom fields (#75579)"
+    (testing "Iceberg partition transform rows in DESCRIBE output shouldn't create duplicate Fields (#75579)"
       (mt/with-dynamic-fn-redefs [athena/run-query (constantly iceberg-partitioned-schema)]
         (is (= expected-fields
                (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))
-    (testing "Hive-style partition sections continue to be deduplicated"
+    (testing "partition columns repeated Hive-style with their real types are only synced once"
       (mt/with-dynamic-fn-redefs [athena/run-query (constantly hive-partitioned-schema)]
         (is (= expected-fields
+               (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))
+    (testing "repeated partition rows are dropped even if the partition section marker is missing"
+      (mt/with-dynamic-fn-redefs [athena/run-query (constantly [{:_col0 "weather_id\tstring\t"}
+                                                                {:_col0 "subcounty_id\tstring\t"}
+                                                                {:_col0 "weight\tfloat\t"}
+                                                                {:_col0 "unused_index\tint\t"}
+                                                                {:_col0 "temporal_key\tstring\t"}
+                                                                {:_col0 "weather_id\tidentity\t"}
+                                                                {:_col0 "temporal_key\tidentity\t"}])]
+        (is (= expected-fields
                (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))))
+
+(deftest ^:parallel describe-unknown-database-type-test
+  (testing "unknown database types should fall back to :type/* instead of a nil base type (#75579)"
+    (mt/with-dynamic-fn-redefs [athena/run-query (constantly [{:_col0 "id\tint\t"}
+                                                              {:_col0 "hll\thyperloglog\t"}])]
+      (is (= #{{:name "id", :base-type :type/Integer, :database-type "int", :database-position 0}
+               {:name "hll", :base-type :type/*, :database-type "hyperloglog", :database-position 1}}
+             (#'athena/describe-table-fields-with-nested-fields "test" "test" "test"))))))
 
 (deftest ^:parallel describe-table-fields-with-nested-fields-test
   (driver/with-driver :athena

--- a/modules/drivers/athena/test/metabase/driver/athena_test.clj
+++ b/modules/drivers/athena/test/metabase/driver/athena_test.clj
@@ -59,6 +59,49 @@
              {:name "ts", :base-type :type/Text, :database-type "STRING", :database-position 1}}
            (#'athena/describe-table-fields-without-nested-fields :athena "test" "test" upper-case-schema-columns)))))
 
+(def ^:private iceberg-partitioned-schema
+  ;; DESCRIBE output for an Iceberg table partitioned by identity(weather_id) and
+  ;; identity(temporal_key). Unlike Hive tables, where the partition section repeats the columns
+  ;; with their real types, Iceberg lists the partition *transform* (e.g. `identity`) in the type
+  ;; column.
+  [{:_col0 "weather_id\tstring\t"}
+   {:_col0 "subcounty_id\tstring\t"}
+   {:_col0 "weight\tfloat\t"}
+   {:_col0 "unused_index\tint\t"}
+   {:_col0 "temporal_key\tstring\t"}
+   {:_col0 "# Partition Information\t\t"}
+   {:_col0 "# col_name\ttransform_type\t"}
+   {:_col0 "weather_id\tidentity\t"}
+   {:_col0 "temporal_key\tidentity\t"}])
+
+(def ^:private hive-partitioned-schema
+  ;; The same table as a Hive-style partitioned table: the partition section repeats the partition
+  ;; columns with their real types.
+  [{:_col0 "weather_id\tstring\t"}
+   {:_col0 "subcounty_id\tstring\t"}
+   {:_col0 "weight\tfloat\t"}
+   {:_col0 "unused_index\tint\t"}
+   {:_col0 "temporal_key\tstring\t"}
+   {:_col0 "# Partition Information\t\t"}
+   {:_col0 "# col_name\tdata_type\tcomment"}
+   {:_col0 "weather_id\tstring\t"}
+   {:_col0 "temporal_key\tstring\t"}])
+
+(deftest describe-iceberg-partitioned-table-test
+  (let [expected-fields #{{:name "weather_id", :base-type :type/Text, :database-type "string", :database-position 0}
+                          {:name "subcounty_id", :base-type :type/Text, :database-type "string", :database-position 1}
+                          {:name "weight", :base-type :type/Float, :database-type "float", :database-position 2}
+                          {:name "unused_index", :base-type :type/Integer, :database-type "int", :database-position 3}
+                          {:name "temporal_key", :base-type :type/Text, :database-type "string", :database-position 4}}]
+    (testing "Iceberg partition-spec rows in DESCRIBE output must not become phantom fields (#75579)"
+      (mt/with-dynamic-fn-redefs [athena/run-query (constantly iceberg-partitioned-schema)]
+        (is (= expected-fields
+               (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))
+    (testing "Hive-style partition sections continue to be deduplicated"
+      (mt/with-dynamic-fn-redefs [athena/run-query (constantly hive-partitioned-schema)]
+        (is (= expected-fields
+               (#'athena/describe-table-fields-with-nested-fields "test" "test" "test")))))))
+
 (deftest ^:parallel describe-table-fields-with-nested-fields-test
   (driver/with-driver :athena
     (is (= #{{:name "id",          :base-type :type/Integer, :database-type "int",    :database-position 0}

